### PR TITLE
#167190422 implements reading statistics

### PR DESCRIPTION
**Pull request by obayomi96**
#### What does this PR do?
- Implements User's Article reading statistics

#### Description of Task to be completed?
- Write test
- Let User's know how many times an article has been read
- Let Author know how many articles he/she has read

#### How should this be manually tested?
- Clone the repo
- Run migrations with;
  `npm run migrate`
- Start the server
 `npm run dev`
- Go to the endpoint below to get a single article with a logged-in user
 `api/v1/articles/:slug`
- Go to the endpoint to know how many article's you've read
 `api/v1/user/readStat`

#### Any background context you want to provide?
- To test if an article has been read when you read it you have to be an authenticated user.

#### What are the relevant pivotal tracker stories?
#167190422

#### Screenshots (if appropriate)
Screenshot 2019-08-13 at 6 50 05 PM

Showing how many article's a user has read

Screenshot 2019-08-13 at 6 50 39 PM

Showing the readCound of an article

### CODE REVIEW

**Review by chuxmykel — APPROVED** [review]
LGTM

**Review by encodedBicoding — APPROVED** [review]
I have no issues with this implementation, Nice one Martins!

**Review by chiboycalix — APPROVED** [review]
Nice work martins. LGTM

**Review by Lundii — COMMENTED** [review]
Nice work Martins, I like what you did by allowing a user get the number of articles, that's an extra feature you added. I would like to get clarifications on one things. If an admin decides to get an article, will it be counted that the article was read? Also, although not necessary, it would be really nice if we can get the articles read by the user not just the count.

**Comment by obayomi96** [comment]
> Nice work Martins, I like what you did by allowing a user to get the number of articles, that's an extra feature you added. I would like to get clarifications on one thing. If an admin decides to get an article, will it be counted that the article was read? Also, although not necessary, it would be really nice if we can get the articles read by the user, not just the count.

Thank you very much for that question @Lundii . Yes If an Admin views an article, It will be recorded that the article has been read one more time as long as the reader is online. The other feature you mentioned is a good idea which I will look into in the future.

**Comment by Lundii** [comment]
> > Nice work Martins, I like what you did by allowing a user to get the number of articles, that's an extra feature you added. I would like to get clarifications on one thing. If an admin decides to get an article, will it be counted that the article was read? Also, although not necessary, it would be really nice if we can get the articles read by the user, not just the count.
> 
> Thank you very much for that question @Lundii . Yes If an Admin views an article, It will be recorded that the article has been read one more time as long as the reader is online. The other feature you mentioned is a good idea which I will look into in the future.

Alright that's great.

**Review by Lundii — APPROVED** [review]

File: server/tests/readStatistics.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  });
+
+  it('should return user read statistics', (done) => {
+    chai
+      .request(app)
+      .get(`${baseUrl}/user/readStat`)
+      .set('Authorization', `Bearer ${userToken}`)
+      .end((err, res) => {
+        const { status } = res.body;
+        expect(status).to.equal(200);
```
Comment (Hadeneekeh) [inline server/tests/readStatistics.test.js:40]: Great job from you Martins, please improve the test by also checking other properties in the response. For instance, properties to expect in the response body, doing that will make it easy to track any intentional or accidental change made to the response returned.

Comment (obayomi96) [reply server/tests/readStatistics.test.js]: Thanks for the feedback, but the line is totally covered %100 and cannot be improved in any way, did you make sure to pull this branch to test the feature and run test locally at all? Please ensure to do that.

Comment (obayomi96) [reply server/tests/readStatistics.test.js:40]: Cool

**Pull request merged**